### PR TITLE
SM-287 / SM-290

### PR DESCRIPTION
**Pull request by lauradeng23**
# Proposed Changes

### CODE REVIEW

File: internal/infrastructure/db/migrations/20231113225121_create_template_dd.sql [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+CREATE TABLE IF NOT EXISTS template_device_definitions (
+    id BIGSERIAL,
+    device_definition_id char(27) NOT NULL,
+    device_style_id char(27),
+    template_name TEXT REFERENCES templates(template_name) NOT NULL,
+    created_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    updated_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP,
+
+    CONSTRAINT template_device_definitions_pkey PRIMARY KEY (id),
+    CONSTRAINT device_def_unique UNIQUE (device_definition_id, template_name)
```
Comment (JamesReate) [inline internal/infrastructure/db/migrations/20231113225121_create_template_dd.sql:14]: also include style id

File: internal/controllers/device_config_controller.go [hunk 2]
```diff
@@ -386,6 +392,11 @@ func (d *DeviceConfigController) GetConfigURLsFromVIN(c *fiber.Ctx) error {
 		ud = &pb.UserDevice{
 			DeviceDefinitionId: definitionResp.DeviceDefinitionId,
 		}
+		if len(definitionResp.DeviceStyleId) > 0 {
+			ud.DeviceStyleId = &definitionResp.DeviceStyleId
+		}
+		// todo: get powertrain type from definition response and include in ud.PowerTrainType
```
Comment (lauradeng23) [inline internal/controllers/device_config_controller.go:398]: not quite sure how to do this here, definitionResp doesn't directly have access to powertrain

File: internal/controllers/device_config_controller.go [hunk 3]
```diff
@@ -251,12 +251,10 @@ func (d *DeviceConfigController) retrieveAndSetVehicleInfo(ctx context.Context,
 		return "", "", 0, fmt.Errorf("failed to retrieve device definition for deviceDefinitionId %s: %w", deviceDefinitionID, err)
 	}
 
-	// Extract vehicle information
 	vehicleYear := int(ddResponse.Type.Year)
 	vehicleMake := ddResponse.Type.MakeSlug
 	vehicleModel := ddResponse.Type.ModelSlug
 
-	// Determine and set Powertrain
 	d.setPowerTrainType(ddResponse, ud)
```
Comment (lauradeng23) [inline internal/controllers/device_config_controller.go:258]: I have this line here in retrieveAndSetVehicleInfo, does this work for setting powertrain?

**Comment by JamesReate** [comment]
overall quite good, specially with tests. I made some tweaks to better handle certain edge cases, and added another couple tests.
